web_monitoring/diffing_server.py

```diff
@@ -97,7 +129,17 @@ def get(self, differ):
 
         # Pass the bytes and any remaining args to the diffing function.
         executor = concurrent.futures.ProcessPoolExecutor()
-        res = yield executor.submit(caller, func, res_a, res_b, **query_params)
+        try:
+            res = yield executor.submit(caller, func, res_a, res_b, **query_params)
+        except KeyError:
+            exceptionMessage = str(sys.exc_info()[1])
+            exceptionMessage = exceptionMessage[1:-1]
+            self.send_error(400, reason="Malformed request. "+exceptionMessage)
+            return
+        except UndecodableContentError:
+            exceptionMessage = str(sys.exc_info()[1])
+            self.send_error(422, reason=exceptionMessage)
+            return
```

Comment:
These errors should already be handled in [`write_error()`](https://github.com/edgi-govdata-archiving/web-monitoring-processing/blob/master/web_monitoring/diffing_server.py#L107-L125). Was that not working for your case?

`write_error` does some important extra work, like:

- differentiating between errors that should be publicly visible and errors that should not (because that could leak security-sensitive information),
- including stack traces _only_ when the server is configured to do so (again, this is a big security concern), and
- formatting the errors in a JSON structure (all output should be JSON here, though we are not perfect at that yet).

Reply:
You are absolutely right about this. I think I was just too eager to handle everything, saw this in the stack trace and went ahead and re-handled it without checking the code first. I'm sorry about this.